src/main/java/com/google/planningworks/cdap/plugins/source/AnaplanExportConfig.java

```diff
@@ -0,0 +1,506 @@
+/*
+ * Copyright © 2021 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.planningworks.cdap.plugins.source;
+
+import com.anaplan.client.AnaplanService;
+import com.google.cloud.ServiceOptions;
+import com.google.common.base.Strings;
+import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;
+import com.google.planningworks.cdap.plugins.base.AnaplanPluginConfig;
+import io.cdap.cdap.api.annotation.Description;
+import io.cdap.cdap.api.annotation.Macro;
+import io.cdap.cdap.api.annotation.Name;
+import io.cdap.cdap.api.data.schema.Schema;
+import io.cdap.cdap.etl.api.FailureCollector;
+import io.cdap.plugin.common.Constants;
+import io.cdap.plugin.common.IdUtils;
+import io.cdap.plugin.format.FileFormat;
+import io.cdap.plugin.format.charset.fixedlength.FixedLengthCharset;
+import io.cdap.plugin.format.plugin.AbstractFileSourceConfig;
+import io.cdap.plugin.format.plugin.FileSourceProperties;
+import java.lang.reflect.Type;
+import java.util.Collections;
+import java.util.Map;
+import java.util.regex.Pattern;
+import javax.annotation.Nullable;
+
+/**
+ * The config class for {@link AnaplanExport} that contains all properties that need to be filled in
+ * by the user when building a pipeline.
+ */
+public class AnaplanExportConfig extends AnaplanPluginConfig implements FileSourceProperties {
```

Comment:
Noted. Removed. I got it from the existing GCS plugin.

Reply:
Any suggestion on this? If we have to use `FileSourceProperties`, then the most of the properties that are suggested to be removed, cannot be removed. How about to make the with the default value and invisible to the user?
Thanks!

Reply:
you don't really need to extend AbstractFileSource. That thing has a lot of code around handling formats, adding paths to the output records, auto-schema detection, etc that isn't relevant to this plugin.

I suggest that you just use Hadoop's CombineTextInputFormat directly. Basically, in prepareRun(), instead of calling super.prepareRun(), do something like:

```
Map<String, String> properties = getFileSystemProperties();
properties.put(FileInputFormat.INPUT_DIR, pathToGCS);
context.setInput(Input.of(config.getReferenceName(), new InputFormatProvider() {
      @Override
      public String getInputFormatClassName() {
        return CombineTextInputFormat.class.getName();
      }

      @Override
      public Map<String, String> getInputFormatConfiguration() {
        return properties;
      }
    }))
```

Reply:
Alternatively, if that's too much refactoring, you can keep this as implementing FileSourceProperties, but return the corresponding constant for all the methods. For example, getPathField() and getFilePattern() should return null instead of returning a value based on user provided config properties.

Reply:
Thanks for the instruction! Going with the second solution.